common/tools/dev-tool/src/config/rollup.base.config.ts

```diff
@@ -16,49 +19,115 @@ interface PackageJson {
   devDependencies: Record<string, string>;
 }
 
+/**
+ * The default sourcemaps plugin does not provide very much information in warnings, so this shim allows us to capture
+ * the active sourcemaps loading context.
+ *
+ * This allows us to selectively disable warnings about missing source maps, for example in core-asynciterator-polyfill.
+ */
+export function sourcemapsExtra() {
+  const _sourcemaps = sourcemaps();
+
+  const load = _sourcemaps.load;
+
+  if (!load) return _sourcemaps;
+
+  return Object.assign(_sourcemaps, {
+    load(this: PluginContext, id: string) {
+      const shim = new Proxy(this, {
```

Comment:
If I understand this correctly, you're replacing a method (load) on the sourcemaps object and using a Proxy to provide a virtual `this` argument that intercepts accessing the `warn` property in order to tag the warnings with an id?

Reply:
Precisely.